enums.rb

```diff
@@ -0,0 +1,94 @@
+# rubocop:disable Style/NestedTernaryOperator, Style/MultilineTernaryOperator, Style/ParallelAssignment, Style/IfUnlessModifier, Style/ExplicitBlockArgument, Style/For, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each { |n| arr.push(n) if yield n }
+    arr
+  end
+
+  def my_all?(*args)
+    case args.size
+    when 0
+      block_given? ? my_select { |n| yield n }.size == size : false
+    when 1
+      args.first.is_a?(Regexp) ?
+      my_select { |n| n.match(args.first) }.size == size :
+      my_select { |n| n.is_a?(args.first) }.size == size
+    end
+  end
+
+  def my_any?(*args)
+    case args.size
+    when 0
+      block_given? ? my_select { |n| yield n }.size.positive? : false
+    when 1
+      args.first.is_a?(Regexp) ?
+      my_select { |n| n.match(args.first) }.size.positive? :
+      my_select { |n| n.is_a?(args.first) }.size.positive?
+    end
+  end
+
+  def my_none?(*args)
+    case args.size
+    when 0
+      block_given? ? !my_select { |n| yield n }.size.positive? : !my_select { |n| n == true }.size.positive?
+    when 1
+      args.first.is_a?(Regexp) ?
+      !my_select { |n| n.match(args.first) }.size.positive? :
+      !my_select { |n| n.is_a?(args.first) }.size.positive?
+    end
+  end
```

Comment:
- #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern. It's throwing a typeError: `class or module required`. You did well with the `Regexp`. You also need to check or test for when a class is given.
- #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern. It's also throwing a typeError. The same solution applies.